### DISCUSSION TITLE
2010/11/04/sexprimer-regulierement-partie-3/

### DISCUSSION THREAD

**Discussion by bric3** [bric3/bric3.github.io#29] [Blog Comments]
# 2010/11/04/sexprimer-regulierement-partie-3/
S&#8217;exprimer régulièrement (Partie 3)
https://blog.arkey.fr/2010/11/04/sexprimer-regulierement-partie-3/

**Comment by bric3** [comment]
Originally posted by Geddons on 2012-09-06T12:37:01Z

----

Bonjour,

Merci beaucoup pour ces articles concernant les regex, je les trouve particulièrement bien dosés et bien adaptés.

Néanmoins, j'ai des corrections à proposer pour la partie LookAhead:

1. Il faut dédoubler vos backslash dans les string d'expressions régulières (\\w =\> \\\\w)   
2. Le dernier snippet de code juste avant les LookBehind ne fera pas un matching sur "private" le deuxième mot de la string" mais bien sur "static" (le premier mot).

**Reply by bric3** [reply]
Merci :)

Effectivement il y a probablement des problèmes, j'ai changé d'outillage pour la coloration syntaxique, et j'ai découvert un paquet de problèmes de caractères dans tous mes snippet de code (caractères HTML, suppréssion de backslashs (tous), etc.), j'ai bien tenté de corriger ça au maximum, mais il y en a qui sont passé entre les mailles du filet. \<del datetime="2012-09-06T13:12:39+00:00"\>Je vais y rejeter un oeil.\</del\> Done, j'ai corrigé aussi les exemples.

**Comment by LeoStileriov** [comment]
Le débutant que je suis s'y perd beaucoup, mais c'est intéressant. 
Y a-t-il quelque part des explications pour comment ajouter l'attribut couleur à une recherche, car je souhaite effectuer la recherche d'un signe en couleur dans un texte. Ma formule est (?<=\s)\s et je voudrais préciser qu'il faut rechercher une espace rouge.

**Reply by bric3** [reply]
> un signe en couleur

Hello, 
Je ne suis pas sûr de comprendre le contexte. Dans du texte simple il n'y a pas de couleur, par contre peut-être est-ce à propos des séquences de code points qu'on peut trouver dans unicode?

**Comment by LeoStileriov** [comment]
J'ai effectué quelques recherches et me suis aperçu que d'une part la fonction recherche/remplacement ne peut rechercher une couleur, et que d'autre part suite à une recherche on ne peut remplacer. Du moins avec l'outil GREP que propose InDesign. Sujet clos donc. Merci de m'avoir répondu, et bonne journée à vous.

**Reply by bric3** [reply]
De rien, bonne journée !
